File: FileBrowser.Contracts/Models/FileSystemItem.cs
using System.ComponentModel.DataAnnotations;

namespace FileBrowser.Contracts.Models
{
    public class FileSystemItem
    {
        [Required]
        public string Name { get; set; } = string.Empty;
        
        [Required]
        public string Path { get; set; } = string.Empty;
        
        public bool IsDirectory { get; set; }
        
        public long? Size { get; set; }
        
        public DateTime? LastModified { get; set; }
        
        public string? Extension { get; set; }
    }
}

File: FileBrowser.Web/wwwroot/js/models/FileSystemItem.js
/**
 * File System Item Model
 * Utility functions for working with file system items
 */

/**
 * Create a FileSystemItem object from API response data
 * @param {Object} data - API response data
 * @returns {Object} FileSystemItem object
 */
function createFileSystemItem(data = {}) {
  const item = {
    name: data.name || "",
    path: data.path || "",
    isDirectory: data.isDirectory || false,
    size: data.size || null,
    lastModified: data.lastModified || null,
    extension: data.extension || null,
  };

  // Add computed properties
  Object.defineProperty(item, "isFile", {
    get: function () {
      return !this.isDirectory;
    },
  });

  return item;
}

// Export for use in other modules
if (typeof module !== "undefined" && module.exports) {
  module.exports = { createFileSystemItem };
} else {
  window.FileSystemItem = { createFileSystemItem };
}
